admiral/pkg/clusters/admiralDatabaseClient.go

```diff
@@ -93,13 +136,90 @@ func NewAdmiralDatabaseClient(admiralConfigPath string, dynamoClientInitFunc fun
 		return nil, fmt.Errorf("error unmarshalling admiral config file, err: %v", err)
 	}
 
-	workloadDatabaseClient.database = &admiralConfig.WorkloadDatabase
-	workloadDatabaseClient.dynamoClient, err = dynamoClientInitFunc(
+	return admiralConfig, nil
+}
+
+func NewDynamicConfigDatabaseClient(path string, dynamoClientInitFunc func(role string, region string) (*DynamoClient, error)) (*DynamicConfigDatabaseClient, error) {
+	var dynamicConfigClient = DynamicConfigDatabaseClient{}
+
+	admiralConfig, err := ReadDynamoConfigForDynamoDB(path)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling admiral config file, err: %v", err)
+	}
+
+	dynamicConfigClient.database = &admiralConfig.DynamicConfigDatabase
+	dynamicConfigClient.database.TableName = common.GetAdmiralParams().DynamicConfigDynamoDBTableName
+	dynamicConfigClient.dynamoClient, err = dynamoClientInitFunc(
 		admiralConfig.WorkloadDatabase.Role,
 		admiralConfig.WorkloadDatabase.Region,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("unable to instantiate dynamo client, err: %v", err)
+		return &dynamicConfigClient, fmt.Errorf("unable to instantiate dynamo client for DynamicConfig, err: %v", err)
+	}
+	return &dynamicConfigClient, nil
+}
+
+func UpdateASyncAdmiralConfig(dbClient AdmiralDatabaseManager, syncTime int) {
+
+	for range time.Tick(time.Minute * time.Duration(syncTime)) {
+		ReadAndUpdateSyncAdmiralConfig(dbClient)
+	}
+}
+
+func ReadAndUpdateSyncAdmiralConfig(dbClient AdmiralDatabaseManager) error {
+
+	dbRawData, err := dbClient.Get("EnableDynamicConfig", common.Admiral)
+	if err != nil {
+		log.Errorf("Error getting EnableDynamicConfig admiral config, err: %v", err)
+		return err
+	}
+
+	configData, ok := dbRawData.(DynamicConfigData)
+	if !ok {
+		return errors.New("Failed to parse DynamicConfigData")
+	}
+
+	if IsDynamicConfigChanged(configData) {
+		log.Infof("Updating DynamicConfigData with Admiral config")
+		UpdateSyncAdmiralConfig(configData)
+	} else {
+		log.Infof("No need to update DynamicConfigData")
+	}
+
+	return nil
+}
+
+func IsDynamicConfigChanged(config DynamicConfigData) bool {
+
+	if config.EnableDynamicConfig != common.Admiral {
+		return false
+	}
+
+	if DynamicConfigCheckSum == sha256.Sum256([]byte(fmt.Sprintf("%v", config))) {
+		return false
+	} else {
+		DynamicConfigCheckSum = sha256.Sum256([]byte(fmt.Sprintf("%v", config)))
+		return true
+	}
+}
+
+func UpdateSyncAdmiralConfig(configData DynamicConfigData) {
```

Comment:
Wouldnt this call update every 10 mins fo all the clusters in the list and not just the one that added ?